src/services/riotApiClient.js

```diff
@@ -109,27 +114,68 @@ class RiotApiClient {
       return { 
         matchDate: match.info.game_datetime,
         matchLength: match.info.game_length,
-        participants: match.metadata.participants
+        matchId: match.metadata.match_id
       }
     })
   
-    matches.forEach (async match => {
-      match.xparticipants = await Promise.all(match.participants.map(participant => this.fetchTFTSummonerInfoByPuuid(participant)))
-    })   
     return matches
     
   }
+
+  async transformMatchData(matchInfo) {
+    const summoners = await Promise.all(matchInfo.info.participants.puuid.map( puuid => this.fetchTFTSummonerInfoByPuuid(puuid)))
+      return {
+      matchId: matchInfo.metadata.match_id,
+      matchTime: matchInfo.info.game_datetime,
+      matchLength: matchInfo.info.game_length,
+      matchParticipants: summoners
+      }
+    return newMatchInfo
+  }
   
   transformSummonerInfo(summonerInfo) {
     const iconID = summonerInfo.profileIconId
-    const url = `http://ddragon.leagueoflegends.com/cdn/${this.latestLolPatchVersion}/img/profileicon/${iconID}.png`;
+    const url = `http://ddragon.leagueoflegends.com/cdn/${this.latestLolPatchVersion}/img/profileicon/${iconID}.png`
     return {
-      name: summonerInfo.name,
+      summonerName: summonerInfo.name,
       level: summonerInfo.summonerLevel, 
-      profileIconUrl: summonerInfo.profileIconId,
+      profileIconUrl: url,
       puuid: summonerInfo.puuid
     }
   }
+  
+  async correctMatchTransformation(matchInfo) {
+    const relevantMatchInfo = await Promise.all(matchInfo.info.participants.map( async info => {
+      return {
+        summoner: await this.fetchTFTSummonerInfoByPuuid(info.puuid),
+        placement: info.placement,
+        level: info.level,
+        playersEliminated: info.players_eliminated,
+        totalDamage: info.total_damage_to_players,
+        champions: info.units.map(championInfo => {
+          const champion = championInfo.character_id.slice(5)
```

Comment:
This is close, but based upon my research, we will have to do something slightly different to return the champion image URL. For now, this is fine.